packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -627,7 +627,9 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
   };
 
   private _onClick = (ev: React.MouseEvent<HTMLDivElement | HTMLAnchorElement | HTMLButtonElement>) => {
-    if (!this.props.disabled && this.props.onClick !== undefined) {
+    if (!this.props.disabled && this.props.menuProps) {
+      this._onMenuClick(ev);
+    } else if (!this.props.disabled && this.props.onClick !== undefined) {
```

Comment:
Similar comment here about rewriting to reduce the need for multiple this.props.disabled checkes